bot/bot.go

```diff
@@ -181,12 +181,20 @@ func (bot *Bot) handle(data []byte) *pb.BotResponse {
 
 	// See if we need to challenge the last move
 	valid := true
+	isWordSmog := g.Rules().Variant() == game.VarWordSmog || g.Rules().Variant() == game.VarWordSmogSuper
 	if g.LastEvent() != nil &&
 		g.LastEvent().Type == pb.GameEvent_TILE_PLACEMENT_MOVE {
 		for _, word := range g.LastWordsFormed() {
-			if !g.Lexicon().HasWord(word) {
-				valid = false
-				break
+			if !isWordSmog {
```

Comment:
this should ideally just be `if g.IsValidWord(word)`; the bot should not duplicate the per-variant rules for whether a move is valid